pandas/tests/copy_view/test_internals.py

```diff
@@ -100,15 +100,15 @@ def test_switch_options():
 @pytest.mark.parametrize(
     "locs, arr",
     [
-        ([0], np.array([-1, -2, -3])),
-        ([1], np.array([-1, -2, -3])),
-        ([5], np.array([-1, -2, -3])),
-        ([0, 1], np.array([-1, -2, -3])),
-        ([0, 2], np.array([-1, -2, -3])),
-        ([0, 1, 2], np.array([-1, -2, -3])),
-        ([1, 2], np.array([-1, -2, -3])),
-        ([1, 3], np.array([-1, -2, -3])),
-        ([1, 3], np.array([[-1, -2, -3], [-4, -5, -6]]).T),
+        ([0], np.array([-1, -2, -3], dtype=np.intp)),
+        ([1], np.array([-1, -2, -3], dtype=np.intp)),
+        ([5], np.array([-1, -2, -3], dtype=np.intp)),
+        ([0, 1], np.array([-1, -2, -3], dtype=np.intp)),
```

Comment:
cc @lithomas1 I don't think that this is valid or can be reached in this way. I think the dimension of your values has to be the same as your indexer. This raises if blk.should_store_value is False, because it does not get broadcast.

Reply:
You're right. Even for the non CoW, the iset operation doesn't fail(strangely), but printing the DataFrame afterwards does.

Reply:
Yep you get a block that has dimension 2 according to block placement but the underlying array has only one column, doesn’t really error when creating but as soon as you access it youll get into trouble :)